crates/cxx-qt-lib-extras/build.rs

```diff
@@ -0,0 +1,54 @@
+// SPDX-FileCopyrightText: 2024 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Laurent Montel <laurent.montel@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx_qt_build::CxxQtBuilder;
+use qt_build_utils::QtBuild;
+
+fn main() {
+    let mut builder = CxxQtBuilder::new();
+
+    // Find the Qt version and tell the Rust compiler
+    // this allows us to have conditional Rust code
+    //
+    // TODO: is this useful to have in cxx-qt-build?
+    println!(
+        "cargo:rustc-cfg=qt_version_major=\"{}\"",
+        QtBuild::new(vec!())
+            .expect("Could not find Qt installation")
+            .version()
+            .major
+    );
+
+    let rust_bridges = vec![
+        "core/qelapsedtimer",
+        "core/qcommandlineoption",
+        "core/qcommandlineparser",
+    ];
+
+    for rust_source in &rust_bridges {
+        builder = builder.file(format!("src/{rust_source}.rs"));
+    }
+
+    let cpp_files = vec![
+        "core/qelapsedtimer",
+        "core/qcommandlineoption",
+        "core/qcommandlineparser",
+    ];
+
+    builder = builder.cc_builder(move |cc| {
+        for cpp_file in &cpp_files {
+            cc.file(format!("src/{cpp_file}.cpp"));
+            println!("cargo:rerun-if-changed=src/{cpp_file}.cpp");
+        }
+        cc.file("src/qt_types.cpp");
+        println!("cargo:rerun-if-changed=src/qt_types.cpp");
+    });
+    println!("cargo:rerun-if-changed=src/assertion_utils.h");
+
+    builder
+        .with_opts(cxx_qt_lib_headers::build_opts())
```

Comment:
wonder what happens if apps just depend on `cxx-qt-lib-extras` and not `cxx-qt-lib` ? Does it end up doing the right thing ? Or do we need to add the `cxx-qt-lib` headers somehow to the `build_opts` ? But then they would need to be in separate folder so it'd need to return a list of `CxxQtBuildersOpts` :thinking: 

Or do we always require the developer also to write `cxx_qt_lib_headers::build_opts()` if this is the case?